diff --git a/configure.ac b/configure.ac
index 9b05c30..6eee7cf 100644
--- a/configure.ac
+++ b/configure.ac
@@ -2263,6 +2263,7 @@ AC_ARG_WITH([ssl-dir],
 		if test "x$openssl" = "xno" ; then
 			AC_MSG_ERROR([cannot use --with-ssl-dir when OpenSSL disabled])
 		fi
+		SSL_DIR="$withval"
 		if test "x$withval" != "xno" ; then
 			case "$withval" in
 				# Relative paths
@@ -2326,11 +2327,11 @@ if test "x$openssl" = "xyes" ; then
 		[
 			dnl Check default openssl install dir
 			if test -n "${need_dash_r}"; then
-				LDFLAGS="-L/usr/local/ssl/lib -R/usr/local/ssl/lib ${saved_LDFLAGS}"
+				LDFLAGS="-L$SSL_DIR/lib -R$SSL_DIR/lib ${saved_LDFLAGS}"
 			else
-				LDFLAGS="-L/usr/local/ssl/lib ${saved_LDFLAGS}"
+				LDFLAGS="-L$SSL_DIR/lib ${saved_LDFLAGS}"
 			fi
-			CPPFLAGS="-I/usr/local/ssl/include ${saved_CPPFLAGS}"
+			CPPFLAGS="-I$SSL_DIR/include ${saved_CPPFLAGS}"
 			AC_CHECK_HEADER([openssl/opensslv.h], ,
 			    [AC_MSG_ERROR([*** OpenSSL headers missing - please install first or check config.log ***])])
 			AC_TRY_LINK_FUNC([RAND_add], [AC_DEFINE([HAVE_OPENSSL])],
